### DISCUSSION TITLE
UID field allows upper case letters

### DISCUSSION THREAD

**Discussion by tomolenet** [strapi/strapi#5415] [Legacy Topics]
**Describe the bug**
Quote from [strapi blog](https://strapi.io/blog/beta-19-single-types-uid-field):
> [...]This field **requires to enter a lowercase**, space-free unique id such as about-us. But don't worry about errors, we got your back: the field will first auto-populate with the name you chose for the Single Type, all spaces replaced with hyphens. But you can also decide to manually add a UID field in your Single Type, and link it to the field of your choice.[...]

But I am able to create an uppercase UID.

**Steps to reproduce the behavior**
1. Create a Content Type with an UID field.
2. Add a Content Type to your collection with an upper case UID.

**Expected behavior**
Expect to automatic lower case the UID or show an error to the user.

**Screenshots**
![image](https://user-images.githubusercontent.com/38924650/75909597-ae1ec900-5e4c-11ea-95bf-88f4729475fe.png)

**System**
- Node.js version: v10.18.1
- NPM version: 6.13.4
- Strapi version: 3.0.0-beta.19.1
- Database: mongodb
- Operating system: linux

**Comment by derrickmehaffy** [comment]
I agree the UID should be all lowercase

**Comment by robertpiosik** [comment]
@derrickmehaffy The UID field value, when created based on another field (such as title) is turned lower case automatically. When you type a value by hand the system assumes you're doing it intentionally. In the web, the url path is case sensitive so it doesn't bring any issue here. What's your use case?

**Comment by tomolenet** [comment]
> @derrickmehaffy The UID field value, when created based on another field (such as title) is turned lower case automatically. When you type a value by hand the system assumes you're doing it intentionally. In the web, the url path is case sensitive so it doesn't bring any issue here. What's your use case?

In my opinion its just an unexpected behavior. Why would you automatically lower case the uuid if the user can change it anyway? In this case you want to force the user to use lower case uuid's imho.

An example use case is if you use the uuid in some frontends to identify an object key for example.

And if this behavior is intended there is at least a mistake in the [blog entry](https://strapi.io/blog/beta-19-single-types-uid-field).

**Comment by robertpiosik** [comment]
I wanna add to the discussion fact that WordPress post's permalink logic forces "slugs" to be lowercase.

**Comment by TakCastel** [comment]
Just here to add my opinion to the discussion I just spent 1 hour trying to understand why my front code didn't work and figured out the API returned all my fields with uppercase. There should be no capital letters in the data returned :/

**Comment by abdonrd** [comment]
I'm also interested in have some options here. Because I need to remove the `-` in some cases.
